plugins/inventory/azure_rm.py

```diff
@@ -223,8 +224,9 @@ def parse(self, inventory, loader, path, cache=True):
             raise
 
     def _credential_setup(self):
+        auth_source = environ.get('ANSIBLE_AZURE_AUTH_SOURCE', None) or self.get_option('auth_source')
```

Comment:
@kingsleyadam I think this order should be adjusted, because the auth_source of the configuration is fetched first, and if the fetch is auto, the environment variable is fetched, thanks!